app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js

```diff
@@ -0,0 +1,537 @@
+                aNode.insertAdjacentHTML('beforeend', '&nbsp;');
+                aNode.appendChild(html.external());
+            } else {
+                if (gitCommit && gitCommit !== null) {
+                    aNode = document.createTextNode(gitCommit);
+                } else {
+                    aNode = html.nonavail();
+                }
+            }
+            html.replaceContent(document.getElementById('git-commit'), aNode);
+
+            // Date.
+            domNode = document.createElement('time');
+            domNode.setAttribute('datetime', createdOn.toISOString());
+            domNode.appendChild(
+                document.createTextNode(createdOn.toCustomISODate()));
+            html.replaceContent(
+                document.getElementById('job-date'), domNode);
+        }
+    }
+
+    function getTests() {
+        var data;
+        var deferred;
+        var labTable;
+
+        function getData(lab) {
+            data.lab_name = lab;
+
+            gLabTable.push(table({
+                tableId: 'lab-table-' + lab,
+                tableLoadingDivId: 'lab-table-loading-' + lab,
+                tableDivId: 'lab-table-div-' + lab
+            }));
+            labTable = gLabTable[gLabTable.length - 1];
+            deferred = request.get('/_ajax/test/suite', data);
+            $.when(deferred)
+                .done(function(response) {
+                    getTestsDone(lab, labTable, response);
+                    getBatchCount(lab, labTable, response);
+                })
+                .fail(error.error, function() {
+                    getTestsFail(lab);
+                });
+        }
+
+        data = {
+            board: gBoard,
+            job: gJob,
+            kernel: gKernel,
+            sort: 'created_on',
+            sort_order: '-1',
+        };
+        // Update the general details
+        deferred = request.get('/_ajax/test/suite', data);
+        $.when(deferred)
+            .done(updateDetails);
+        // Update each lab with the specific data
+        gLabList.forEach(getData);
+    }
+
+    if (document.getElementById('job-name') !== null) {
+        gJob = document.getElementById('job-name').value;
+    }
+    if (document.getElementById('kernel-name') !== null) {
+        gKernel = document.getElementById('kernel-name').value;
+    }
+    if (document.getElementById('board-name') !== null) {
+        gBoard = document.getElementById('board-name').value;
+    }
+    if (document.getElementById('search-filter') !== null) {
+        gSearchFilter = document.getElementById('search-filter').value;
+    }
+    if (document.getElementById('file-server') !== null) {
+        gFileServer = document.getElementById('file-server').value;
+    }
+
+    function extendLabList(lab) {
+        gLabList.push(lab.value)
+    }
+
+    Array.prototype.slice.call(document.querySelectorAll('input.lab-names'))
+        .map(extendLabList);
+    if (gLabList) {
+        if (gLabList.length > 0) {
```

Comment:
You can squeeze this check into just one line, and only check for `gLabList` length.
An empty array in Javascript evaluates to true.